src/components/screen/SignUp.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { ReactElement, useState } from 'react';
+
+import Button from '../shared/Button';
+import { DefaultNavigationProps } from '../../types';
+import { EditText } from '@dooboo-ui/native';
+import { ScrollView } from 'react-native-gesture-handler';
+import StatusBar from '../shared/StatusBar';
+import { View } from 'react-native';
+import { getString } from '../../../STRINGS';
+import styled from 'styled-components/native';
+import { useThemeContext } from '@dooboo-ui/native-theme';
+
+const Container = styled.SafeAreaView`
+  flex: 1;
+  background: ${({ theme }): string => theme.background};
+`;
+
+const Wrapper = styled.View`
+  margin: 44px;
+`;
+
+const ButtonWrapper = styled.View`
+  width: 100%;
+  margin-top: 20px;
+  flex-direction: row;
+`;
+
+interface Props {
+  navigation: DefaultNavigationProps<'SignUp'>;
+}
+
+function Page(props: Props): ReactElement {
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [confirmPassword, setConfirmPassword] = useState<string>('');
+  const [name, setName] = useState<string>('');
+  const [status, setStatus] = useState<string>('');
+
+  const [errorEmail, setErrorEmail] = useState<string>('');
+  const [errorPassword, setErrorPassword] = useState<string>('');
+  const [errorConfirmPassword, setErrorConfirmPassword] = useState<string>('');
+  const [errorName, setErrorName] = useState<string>('');
+  const [isSignUp, setIsSignUp] = useState<boolean>(false);
+  let timer: number;
+
+  const { theme } = useThemeContext();
+
+  const validateEmail = (email: string): boolean => {
+    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(email);
+  };
+
+  const validatePassword = (password: string): boolean => {
```

Comment:
Although this isn't used in other code, you can still define this into utility functions for future reusability.